src/neovim.rs

```diff
@@ -100,8 +106,11 @@ where
       queue: Arc::new(Mutex::new(Vec::new())),
     };
 
-    let req_t = req.clone();
-    let fut = Self::io_loop(handler, reader, req_t);
+    let (sender, receiver) = unbounded();
+    let fut = future::try_join(
+      Self::io_loop(reader, sender, req.clone()),
+      req.clone().handle_rec(handler, receiver)
+    ).map_ok(|_| ());
```

Comment:
Why is this needed? Both futures return () on success, dont they?

Reply:
Exactly! Which means since `future::try_join()` returns the results from both futures, the actual return type is `((), ())` - which is a bit awkward, so we `.map_ok()` it back into a single `()`